packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/TimeRangeForm.tsx

```diff
@@ -86,6 +86,28 @@ export const TimeRangeForm: React.FC<Props> = (props) => {
     [timeZone]
   );
 
+  const onFromChange = useCallback(
+    (event: FormEvent<HTMLInputElement>) => {
+      const newState = eventToState(event, false, timeZone);
+      const invalid = isRangeInvalid(newState, to, timeZone);
+
+      newState.invalid = newState.invalid || invalid;
```

Comment:
nit: instead of doing this you can change the `eventToState` function directly to include your range check.